llvm/include/llvm/BinaryFormat/ELF.h

```diff
@@ -1643,6 +1646,11 @@ enum {
   NT_ANDROID_TYPE_MEMTAG = 4,
 };
 
+// ARM note types
+enum {
+  NT_ARM_TYPE_PAUTH_ABI_TAG = 1,
```

Comment:
I couldn't see this value defined in the linked spec?

Reply:
I think I have a few (small) concerns with this:
1) It's potentially possible (admittedly unlikely) that the name could be specified for something different in the future, so you'd have to modify existing sites to use a different name.
2) Giving it a name means people will search for the string in the docs and not find it, potentially causing confusion in the future.

I don't know who is in charge of those docs, but is it possible to update them to specifically name the NT_* value? That would resolve my concern.

Reply:
@smithp35 Could we provide this constant name `NT_ARM_TYPE_PAUTH_ABI_TAG` (or some other name) and its value `1` in the section 14.1 of the spec https://github.com/ARM-software/abi-aa/blob/main/pauthabielf64/pauthabielf64.rst#appendix-alternative-elf-marking-using-gnu-program-properties? I could submit a PR to https://github.com/ARM-software/abi-aa if you wish.

Reply:
@smithp35 A kind reminder regarding my question above.

Reply:
Just to make sure I've got this right:
The link above is for GNU program properties:
https://github.com/ARM-software/abi-aa/blob/main/pauthabielf64/pauthabielf64.rst#19appendix-alternative-elf-marking-using-gnu-program-properties

It defines an additional program property on top of the existing .note.GNU.property defined in the Linux ABI extensions https://raw.githubusercontent.com/wiki/hjl-tools/linux-abi/linux-abi-draft.pdf that already has its own `NT_GNU_PROPERTY_TYPE_0` 

There is a part here which defines .note.AARCH64-PAUTH-ABI-tag which I think is where an additional `NT_ARM_TYPE_PAUTH_ABI_TAG` would make sense.
https://github.com/ARM-software/abi-aa/blob/main/pauthabielf64/pauthabielf64.rst#141default-marking-schema

Have I got that right?

In any case please can you raise an issue on the AArch64 ABI for that https://github.com/ARM-software/abi-aa so it can be addressed there. I'll also need to update the document to make it compatible with some forthcoming changes in the ABI to do with ELF marking. It is unfortunate that this will land too late for your timescale.

Essentially these pull-requests:
* https://github.com/ARM-software/abi-aa/pull/230 build attributes for relocatable objects
* https://github.com/ARM-software/abi-aa/pull/229 rationale for GNU properties in executable shared libraries

None have been merged yet, but pending more review comments there is a consensus within Arm that we'll be using build attributes (like 32-bit Arm) for ELF metadata in relocatable objects and for SysvAbi platforms such as Linux we'll be using .note.GNU.properties as our reference ELF marking scheme for executables and shared-libraries.

I expect that when the PAuthABI specification comes out of Alpha it will need to be updated, most likely to make the current Alternative ELF marking will be the reference marking, and the .note.AARCH64-PAUTH-ABI-tag will become the alternative. I've already defined build attibutes for PAuthABI that match the existing implementation.

Reply:
> There is a part here which defines .note.AARCH64-PAUTH-ABI-tag which I think is where an additional `NT_ARM_TYPE_PAUTH_ABI_TAG` would make sense. https://github.com/ARM-software/abi-aa/blob/main/pauthabielf64/pauthabielf64.rst#141default-marking-schema
> 
> Have I got that right?

Yes. It looks like I placed link a wrong link but a correct section number above :) I meant 14.1 https://github.com/ARM-software/abi-aa/blob/main/pauthabielf64/pauthabielf64.rst#141default-marking-schema.

Submitted https://github.com/ARM-software/abi-aa/issues/234. Thanks!